### PR TITLE
Added NCM source

### PR DESCRIPTION
**Pull request by jsfaint**

### CODE REVIEW

File: autoload/cm/sources/necosyntax.vim [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+"=============================================================================
+
+let s:initialized = 0
+function! cm#sources#necosyntax#refresh(opt, ctx)
+  if !s:initialized
+    call necosyntax#initialize()
+    let s:initialized = 1
+  endif
+
+  let l:col = a:ctx['col']
```
Comment (Shougo) [inline autoload/cm/sources/necosyntax.vim:14]: Please remove `l:` prefix.

**Comment by Shougo** [comment]
Additional:  Please add the ncm support description in doc/necosyntax.vim.

**Comment by Shougo** [comment]
I have fixed them

**Comment by jsfaint** [comment]
Thanks for your great work on these excellent plugins 👍

**Pull request merged**